mobile/Screen/GalleryScreen.js

```diff
@@ -44,32 +105,66 @@ function GalleryScreen({ navigation }) {
     getAllSubmissions();
   }, []);
 
+  let filteredData = allImageData;
+  if (loadImages && selectedTags.length > 0) {
+    filteredData = filteredData.filter((imageData) => imageData.SubmissionData.tags
+      .some((tag) => tagLabels.includes(tag)));
+  }
+
   return (
-    <ScrollView style={styles.scrollView}>
-      <Text>Gallery Screen</Text>
-      {
-        loadImages ? (
-          <>
-            {
-              allImageData.map((imageData) => (
-                <ArtworkCard
-                  ImageURL={imageData.ImageURL}
-                  id={imageData.SubmissionData._id}
-                  navigation={navigation}
-                />
-              ))
-            }
-          </>
-        ) : <Text>Nothing</Text>
-      }
-    </ScrollView>
+    <SafeAreaView>
+      <View style={styles.header}>
+        <Text style={styles.headerText}>Gallery</Text>
+        <TouchableOpacity onPress={() => navigation.navigate('Artwork Filter')}>
```

Comment:
Currently, when you enable a filter, go back to the Gallery screen, and then go back to the filter screen, none of the pre-selected filters are saved. Maybe use props to make sure that filters that are selected before are kept?